src/Compilers/Core/Portable/Compilation/Compilation.cs

```diff
@@ -62,15 +62,15 @@ public abstract partial class Compilation
         // Protected for access in CSharpCompilation.WithAdditionalFeatures
         protected readonly IReadOnlyDictionary<string, string> _features;
 
-        public ScriptCompilationInfo ScriptCompilationInfo => CommonScriptCompilationInfo;
-        internal abstract ScriptCompilationInfo CommonScriptCompilationInfo { get; }
+        public ScriptCompilationInfo? ScriptCompilationInfo => CommonScriptCompilationInfo;
+        internal abstract ScriptCompilationInfo? CommonScriptCompilationInfo { get; }
 
         internal Compilation(
-            string name,
+            string? name,
             ImmutableArray<MetadataReference> references,
             IReadOnlyDictionary<string, string> features,
             bool isSubmission,
-            AsyncQueue<CompilationEvent> eventQueue)
+            AsyncQueue<CompilationEvent>? eventQueue)
```

Comment:
When is having a `null` value for `eventQueue` used in the code base? I couldn't find any. It would seem like the type would fall over if that happened